src/harness/testcases/WINNF_FT_S_PAT_testcase.py

```diff
@@ -103,32 +172,45 @@ def test_WINNF_FT_S_PAT_1(self, config_filename):
     
     config = loadConfig(config_filename)
     num_tests = len(config)
-    test_pass_threshold = 0.95 #percentage of passed test expected
+    test_pass_threshold = 0.999 #percentage of passed test expected
     num_passed_tests = 0.0
     num_tests = len(config)
     max_fail_num = int ((1-test_pass_threshold) * num_tests)
     num_failed_tests = 0.0
+    num_invalid_tests = 0.0
     for request in  config:
-       refResponse = computePropagationAntennaModel(request)
-       sasResponse = self._sas_admin.QueryPropagationAndAntennaModel(request)
+        try:
+           ref_response = computePropagationAntennaModel(request)
+        except ValueError as e:
+           logging.debug(e)
+           num_invalid_tests += 1.0
+           num_tests -= 1.0
+           max_fail_num = int ((1-test_pass_threshold) * num_tests)
+           continue
+       
+       # don't send a response to sas if invalid
+       sas_response = self._sas_admin.QueryPropagationAndAntennaModel(request)
+
 
        # Check response
        this_test = False
-       if 'pathlossDb' in refResponse and 'txAntennaGainDbi' in refResponse:
-           this_test = (sasResponse['pathlossDb'] < refResponse['pathlossDb'] + 1) and (sasResponse['txAntennaGainDbi'] < (refResponse['txAntennaGainDbi'] + .2))
-               
-           if 'rxAntennaGainDbi' in refResponse:
-               this_test = this_test and (sasResponse['rxAntennaGainDbi'] < (refResponse['rxAntennaGainDbi'] + .2))
-       else:
-           self.assertEqual(sasResponse, refResponse)
-           this_test = True
+       if 'pathlossDb' in sas_response and 'txAntennaGainDbi' in sas_response:
+           this_test = (sas_response['pathlossDb'] < ref_response['pathlossDb'] + 1) and (sas_response['txAntennaGainDbi'] < (ref_response['txAntennaGainDbi'] + .2))
+
+           if 'fss' in request:
+               if 'rxAntennaGainRequired' in request['fss']:
+                   if (request['fss']['rxAntennaGainRequired'] == True) :
+                       if 'rxAntennaGainDbi' in sas_response:
+                           this_test = this_test and (sas_response['rxAntennaGainDbi'] < (ref_response['rxAntennaGainDbi'] + .2))
+                       else:
+                           this_test = False
 
        if this_test:
            num_passed_tests += 1.0
        else:
            num_failed_tests += 1.0
-       
-       if num_failed_tests >= max_fail_num:
+           
+        if num_failed_tests > max_fail_num: #test fails if number of failed tests greater than allowed
```

Comment:
Please use two spaces and remove the preceding line break to make it clear that this is inside the `else`.